### DISCUSSION TITLE
Enabling integration with DigiLocker (and any other similar wallet)

### DISCUSSION THREAD

**Discussion by alok-os** [Sunbird-Ed/Community#246] [Feature request]
Hello @krishnaj20,  @kameshbhr @sharathkashyap @vrayulu @reshmi-nair  @rhwarrier 

Keeping Indian adopters in mind, we should look at enabling integration of Ed with Digilocker; implementing it in such a way so that integration of any other wallet is also possible (with a plug-in adapter). Here is the initial thought on the Digilocker integration:
1. An instance (ex Diksha) of Sunbird Ed can become "Issuer" on Digilocker allowing its Ed instance users to get access to their verifiable digital certificates in their digilocker account
2. A digilocker user should be allowed to "pull" their certificate in their digilocker account. We need to finalize what information user needs to provide to pull their record without compromising data privacy.
3. The digilocker user should be able to share their certificate with a consent to a "requester" on digilocker. 

The solution should be implemented so that other digital wallets can also be integrated in the future. We should look at pluggable adapter model.

@pramodkvarma

**Comment by parthlawate** [comment]
in effect this would mean that a Sunbird RC issued certificate (I assume thats what ED uses) will need to allow the above flow ? In which case should this request move to SB RC forums ?

**Comment by rhwarrier** [comment]
That's right - and from what was indicated, Cowin as an issuer into Digilocker is already allowing them to pull credentials from RC. We'll need to sync with the Digilocker team in order to better understand what needs doing in order to allow Diksha to become an 'issuer', what params will be passed etc. 
Kamesh, Krishna - we'll need some inputs from the Diksha and Ed front as well, to see if there are any specific use cases you'd like to address
